docs/elements.md

```diff
@@ -347,6 +350,71 @@ Attribute | Type | Default | Description
+`grading-method` | string | "ordered" | One of the following: `ordered`, `unordered`, `ranking`, `external`. See more details below.
+`file-name` | string | `user_code.py`  | Name of the file where the information from the blocks will be saved, to be used by the external grader.
+`source-header` | string | "Drag from here" | The text that appears at the start of the source area.
+`solution-header` | string| "Construct your solution here" |  The text that appears at the start of the solution area.
+`solution-placement` | string | "right" | "right" shows the source and solution areas aligned side-by-side. "bottom" shows the solution area below the source area.
+
+Within the `pl-order-blocks` element, each answer block must be specified with a `pl-answer` that has the following attributes:
+
+Attribute | Type | Default | Description
+--- | --- | --- | ---
+`correct` | boolean | true | Specifies whether the answer block is a correct answer to the question (and should be moved to the solution area).
+`ranking` | positive integer | — | This attribute is used when `grading-method="ranking"` and it specifies the correct ranking of the answer block. For example, a block with ranking `2` should be placed below a block with ranking `1`. The same ranking can be used when the order of certain blocks is not relevant. Blocks that can be placed at any position should not have the `ranking` attribute.
+`indent` | integer in [-1, 4] | -1 | Specifies the correct indentation level of the block. For example, a value of `2` means the block should be indented twice. A value of `-1` means the indention of the block does not matter.
+
+#### Details
+
+Different grading options are defined via the attribute `grading-method`:
+
+* `ordered`: in this method, the correct ordering of the blocks is defined by the ordering in which
+the correct answers (defined in `pl-answer`) appear in the HTML file. There is no partial credit for this option.
+* `unordered`: in this method, if `n` is the total number of correct blocks, each correct block moved to the solution area is given `1/n` points, and each incorrect block moved to the solution area is subtracted by `1/n` points. The final score will be at least 0 (the student cannot earn a negative score by only moving incorrect answers). Note the ordering of the blocks does not matter. That is, any permutation of the answers within the solution area are accepted.
+* `ranking`: in this method, the `ranking` attribute of the `pl-answer` options are used to check answer ordering. Every answer block *X* should have a `ranking` integer that is less than or equal to the answer block immediately below *X*. That is, the sequence of `ranking` integers of all the answer blocks should form a *nonstrictly increasing* sequence. If `n` is the total number of answers, each correctly ordered answer is worth `1/n`, up to the first incorrectly ordered answer.
```

Comment:
Please add `There is no partial credit for this option.` to the end of this paragraph, or explain how the partial credit works.

Reply:
fixed; every paragraph now explicitly specify whether partial credit is possible